smart-common/src/main/java/org/smartdata/model/FileInfoUpdate.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartdata.model;
+
+import java.util.Objects;
+
+public class FileInfoUpdate {
+  private String path;
+  private Long length;
+  private Boolean isdir;
```

Comment:
How about using primitives for length, isdir, blocksize and maybe other appropriate fields?

Reply:
The non-null fields of this class are transformed to the `SET field_name = field_value` clause of `file` table update query. If we use primitives, as it was done before, we will update such fields in db with the default values of corresponding primitives, if they weren't set, rewriting the correct original ones